src/service/resources-impl.js

```diff
@@ -1723,7 +1723,7 @@ export class Resource {
       return Promise.resolve();
     }
 
-    if (!this.renderOutsideViewport() && !this.isInViewport()) {
+    if (!this.isInViewport() && !this.renderOutsideViewport()) {
```

Comment:
@jridgewell could you please also respond to this. Essentially I'd like to move to the system where decision yields one of three results:
- Layout now
- Prerender now
- Do nothing

I think this method is the one that would give us necessary data to calculate this signal.